common/scala/src/main/scala/org/apache/openwhisk/core/entity/WhiskEntity.scala

```diff
@@ -115,7 +116,8 @@ object WhiskEntity {
 
 object WhiskDocumentReader extends DocumentReader {
   override def read[A](ma: Manifest[A], value: JsValue) = {
-    ma.runtimeClass match {
+    val entityType = value.asJsObject.fields("entityType").convertTo[String]
```

Comment:
`entityType` is technically an optional attribute as it was introduced later (#3164). So need to treat that as `Option`

Reply:
That’s a good point. If the property 
 doesn’t exist then fall back to the old code.

Reply:
I will consider this idea